### PR TITLE
Simulated inventory, step 3

### PR DESCRIPTION
**Pull request by Jim3Things**
This adds the initial rack simulation, cleans up some of the simpleSM usage, and a few minor other changes.  Once again, there are more steps planned beyond this.

### CODE REVIEW

File: internal/services/inventory/rack.go [hunk 1]
```diff
@@ -50,31 +88,177 @@ func newRack(ctx context.Context, def *pb.ExternalRack) *rack {
...
+		// currently has no status, so this request is always forwarded
+		// to one or more contained elements.
+		addr := body.Target
+
+		switch addr.Element.(type) {
+		case *services.InventoryAddress_Pdu:
+			// The PDU is the one component that we can access without
+			// using the TOR as a network hop.  This reflects an
+			// intentional simplification in the simulation: that the
+			// TOR cannot disconnect the PDU, and hte PDU cannot stop
```
Comment (MikeCJohnsonUk) [inline internal/services/inventory/rack.go:218]: Typo - "hte"

File: internal/sm/simpleSm.go [hunk 2]
```diff
@@ -2,11 +2,55 @@ package sm
...
 
 	"github.com/Jim3Things/CloudChamber/internal/common"
 	"github.com/Jim3Things/CloudChamber/internal/tracing"
 )
 
+// Envelope holds an incoming request to the state machine, along with context
+// needed to process and respond to it.
+type Envelope struct {
+	// CH is the channel that the completion response is to be sent over.
+	CH chan *Response
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply internal/sm/simpleSm.go]: fixed

File: internal/services/inventory/pdu.go [hunk 3]
```diff
@@ -154,60 +155,81 @@ func (s *pduWorking) changePower(
...
 			if !power.Power {
+				sc := trace.SpanFromContext(ctx).SpanContext()
+
 				for i := range p.cables {
 
 					changed, err := p.cables[i].force(false, after.Ticks, occursAt)
 
 					if changed && err == nil {
 						// power is on to this blade.  Turn it off, but tell
 						// the blade to not reply, as this is a side effect.
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply internal/services/inventory/pdu.go]: clarified wording

File: internal/services/inventory/rack.go (hunk 1 above)
Comment (Jim3Things) [reply internal/services/inventory/rack.go]: fixed

**Review by MikeCJohnsonUk — APPROVED** [review]

**Pull request merged**